FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/DataProcessor/DataConverter/CustomSchemaConverter.cs

```diff
@@ -27,24 +27,45 @@ namespace Microsoft.Health.Fhir.Synapse.Core.DataProcessor.DataConverter
     public class CustomSchemaConverter : IDataSchemaConverter
     {
         private readonly JsonProcessor _jsonProcessor;
-        private readonly ITemplateProvider _templateProvider;
         private readonly ILogger<CustomSchemaConverter> _logger;
+        private readonly IContainerRegistryTemplateProvider _containerRegistryTemplateProvider;
+        private readonly string _schemaImageReference;
+
+        private readonly object _templateProviderLock = new object();
+        private ITemplateProvider _templateProvider;
 
         public CustomSchemaConverter(
             IContainerRegistryTemplateProvider containerRegistryTemplateProvider,
             IOptions<SchemaConfiguration> schemaConfiguration,
             ILogger<CustomSchemaConverter> logger)
         {
-            _logger = logger;
-            if (!string.IsNullOrWhiteSpace(schemaConfiguration.Value.SchemaImageReference))
+            EnsureArg.IsNotNull(schemaConfiguration, nameof(schemaConfiguration));
+
+            _logger = EnsureArg.IsNotNull(logger, nameof(logger));
+            _containerRegistryTemplateProvider = EnsureArg.IsNotNull(containerRegistryTemplateProvider, nameof(containerRegistryTemplateProvider));
+            _schemaImageReference = schemaConfiguration.Value.SchemaImageReference;
+
+            _jsonProcessor = new JsonProcessor(new ProcessorSettings());
+        }
+
+        private ITemplateProvider TemplateProvider
+        {
+            get
             {
-                var templateCollections = containerRegistryTemplateProvider.GetTemplateCollectionAsync(
-                    schemaConfiguration.Value.SchemaImageReference,
-                    CancellationToken.None).Result;
+                // Do the lazy initialization.
+                if (_templateProvider is null)
+                {
+                    lock (_templateProviderLock)
+                    {
+                        _templateProvider ??= new TemplateProvider(_containerRegistryTemplateProvider.GetTemplateCollectionAsync(
```

Comment:
Is it possible that new `TemplateProvider` fails, and the `_templateProvider` is null?